imports/plugins/core/orders/client/components/OrderRefunds.js

```diff
@@ -0,0 +1,303 @@
+import React, { Fragment, useEffect, useRef, useState } from "react";
+import PropTypes from "prop-types";
+import { Mutation } from "react-apollo";
+import { Form } from "reacto-form";
+import withStyles from "@material-ui/core/styles/withStyles";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import CardHeader from "@material-ui/core/CardHeader";
+import FormControl from "@material-ui/core/FormControl";
+import InputLabel from "@material-ui/core/InputLabel";
+import Grid from "@material-ui/core/Grid";
+import MenuItem from "@material-ui/core/MenuItem";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+import Select from "@material-ui/core/Select";
+import Typography from "@material-ui/core/Typography";
+import ErrorsBlock from "@reactioncommerce/components/ErrorsBlock/v1";
+import Field from "@reactioncommerce/components/Field/v1";
+import TextInput from "@reactioncommerce/components/TextInput/v1";
+import { i18next, Reaction } from "/client/api";
+import formatMoney from "/imports/utils/formatMoney";
+import ConfirmButton from "/imports/client/ui/components/ConfirmButton";
+import createRefundMutation from "../graphql/mutations/createRefund";
+import OrderPreviousRefunds from "./OrderPreviousRefunds";
+
+const styles = (theme) => ({
+  dividerSpacing: {
+    marginBottom: theme.spacing(4),
+    marginTop: theme.spacing(4)
+  },
+  fontWeightSemiBold: {
+    fontWeight: theme.typography.fontWeightSemiBold
+  },
+  formControl: {
+    minWidth: "250px",
+    width: "250px"
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2)
+  }
+});
+
+/**
+ * @name OrderRefunds
+ * @param {Object} props Component props
+ * @returns {React.Component} returns a React component
+ */
+function OrderRefunds(props) {
+  const hasPermission = Reaction.hasPermission(["reaction-orders", "order/fulfillment"], Reaction.getUserId(), Reaction.getShopId());
+  const { classes, order } = props;
+  const { payments } = order;
+
+  // useRef
+  const inputLabel = useRef(null);
+
+  // useState
+  const [labelWidth, setLabelWidth] = useState(0);
+  const [refundReasonSelectValues, setRefundReasonSelectValues] = useState({ reason: "" });
+  const [refundTotal, setRefundTotal] = useState(0.00);
+
+  // useEffect
+  // update label width when refund select is activate
+  useEffect(() => {
+    if (inputLabel && inputLabel.current) {
+      setLabelWidth(inputLabel.current.offsetWidth);
+    }
+  }, []);
+
+  const canRefund = payments.every((payment) => payment.status !== "refunded");
```

Comment:
The payment method also has a `canRefund` prop, meaning does the method even support refunds. I believe both of our built-in payment methods have this set to `true`, but you could set the example IOU method to `false` and test what happens.

You should be able to get this through GraphQL by requesting `method.canRefund` on each `payments`:

```graphql
{
  payments {
    method {
      canRefund
    }
  }
}
```

It seems like the method not supporting refunds is slightly different from the meaning of `canRefund` here. So you may want to treat them separately and maybe UI should explicitly show a "Payment method does not support refunds" message somewhere? cc @rymorgan

Reply:
Yeah, my `canRefund` here is "is there an amount left to refund, or has it already been fully refunded". I'll refactor that name a bit and implement a check to make sure it's even allowed.

@reactioncommerce/design I have added a few little things on top of your designs, just to provide a little contextual information where i wasn't seeing it, I can add a message if refunds aren't supported in a similar fashion, or we can discuss how to design this differently.

![Order_Details_for_order_reference__JvhbMZkxhsZJqiL8f](https://user-images.githubusercontent.com/4482263/62494392-41ddc900-b788-11e9-9d10-3e808f186ee9.png)

Reply:
Updated to use `canRefund` from each payment as intended and using `isPaymentFullyRefunded` in place of the `canRefund` version I had.